### PR TITLE
Only hash dep node indices of deps of anon tasks

### PR DESCRIPTION
**Pull request by Zoxc**
Another change split out from https://github.com/rust-lang/rust/pull/60035.

r? @michaelwoerister

### CODE REVIEW

File: src/librustc/dep_graph/graph.rs [hunk 1]
```diff
@@ -1021,25 +1021,17 @@ impl CurrentDepGraph {
     fn complete_anon_task(&mut self, kind: DepKind, task_deps: TaskDeps) -> DepNodeIndex {
         debug_assert!(!kind.is_eval_always());
 
-        let mut fingerprint = self.anon_id_seed;
```
Comment (michaelwoerister) [inline src/librustc/dep_graph/graph.rs]: Can you add a comment explaining why this works? Especially why it is OK to stable-hash unstable data like `DepNodeIndex` values?

Comment (Zoxc) [reply src/librustc/dep_graph/graph.rs]: Done.

**Comment by michaelwoerister** [comment]
Thanks, @Zoxc! The queue is enticingly empty right now, so @bors r+

We'll get perf results from the regular build.

**Comment by michaelwoerister** [comment]
Looks like this did not make much of a difference: 
https://perf.rust-lang.org/compare.html?start=55c48b4e823ad3518bb8e6d3fd7da4f2fc0c5fc2&end=c6ac57564852cb6e2d0db60f7b46d9eb98d4b449&stat=instructions:u

**Pull request merged**